Add hasActivePlugin selector

psealock opened the pull request:
### Goal
Find out if a site has a particular active plugin.

### Why
Site stats will now include Store stats for jetpack enabled stores. In order to know if some kind of toggle control should be rendered, we need to know if `woocommerce` plugin is installed and active.

![screen shot 2017-04-10 at 11 52 41 am](https://cloud.githubusercontent.com/assets/1922453/24844315/e3c5647a-1dfd-11e7-8125-55a6e5a18dcf.png)

The parent components's connect `mapStateToProps` will have something like this
```js
isWooConnect: isJetpack && hasActivePlugin( state, siteId, 'woocommerce' ),
```

psealock commented:
Fixes https://github.com/Automattic/wp-calypso/issues/12970

psealock commented:
@blowery is this a sane approach?

samouri commented on client/state/plugins/installed/selectors.js:
```diff
@@ -127,3 +127,24 @@ export function isPluginDoingAction( state, siteId, pluginId ) {
+export function hasActivePlugin( state, siteId, pluginSlug ) {
+	const sitePlugins = state.plugins.installed.plugins[ siteId ];
+	if ( ! sitePlugins ) {
+		return false;
```
is there a need to differentiate between unfetched + actually false?

If so, a common practice i've seen would be to return null

psealock replied:
I wanted to keep the returned value a `Bool`. Maybe I should rename this function to `isPluginActive`.

>is there a need to differentiate between unfetched + actually false?

I'm not entirely following

samouri commented on client/state/plugins/installed/selectors.js:
```diff
@@ -127,3 +127,24 @@ export function isPluginDoingAction( state, siteId, pluginId ) {
+	if ( ! sitePlugins ) {
+		return false;
+	}
+	const plugin = find( sitePlugins, sitePlugin => sitePlugin.slug === pluginSlug );
```
a slightly cooler way of writing this is:
```js
const plugin = find( sitePlugins, { slug: pluginSlug } );
```

psealock replied:
Thanks. I had this before, but felt this way was easier to read because it closely resembles `Array.prototype.find`. If Calypso tends to towards your way, I'm cool with changing.

samouri commented:
Is there a different pattern for creating plugin selectors? I know we are now trying to place selectors inside of `state/selectors/index` but it seems totally reasonable to me that plugins would have a separate pattern.

psealock commented:
>Is there a different pattern for creating plugin selectors? I know for most of calypso we are trying to place selectors inside of `state/selectors/index`

I didn't see that folder. Is the intent to migrate all selectors here? I see many folders in `state/` have their own `selectors.js` file.

psealock commented:
### Update
* Renamed `hasActivePlugin` to `isPluginActive` to reflect the current patterns of selector names starting with "get" or "is".
* Changed use of lodash `find` to more terse syntax.
* Moved the selector to `state/selectors`. The [docs](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#selectors) point out the benefits of accessing the entire state tree.


cc @samouri

blowery commented on client/state/selectors/is-plugin-active.js:
```diff
@@ -0,0 +1,25 @@
+ */
+export default function isPluginActive( state, siteId, pluginSlug ) {
+	const sitePlugins = state.plugins.installed.plugins[ siteId ];
+	if ( ! sitePlugins ) {
```
you don't technically need this check. `find` will handle it for you.

blowery commented on client/state/selectors/test/is-plugin-active.js:
```diff
@@ -0,0 +1,56 @@
+const state = deepFreeze( {
+	plugins: {
+		installed: {
+			isRequesting: {
```
why is this bit of state included in the test?
